openstack/README.md

```diff
@@ -1,38 +1,133 @@
+# Table of Contents
+
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Provision a VM on OpenStack](#provision-a-vm-on-openstack)
+
+# Introduction
+
+This document describes the requirements and the process to execute the provisioning of a Cloud VM on Openstack.
+
 # Prerequisites
 
 You need to have both the `decorator` and `openstacksdk` pip packages installed.
 Depending on the existing state of your machine, you might need to do:
 
-```
+```bash
 [sudo] pip install [--upgrade] decorator
 [sudo] pip install [--upgrade] openstacksdk
 ```
 
+Install openstack.cloud Ansible collection.
+
+```bash
+$ ansible-galaxy collection install openstack.cloud
+```
+
 # Provision a VM on OpenStack
 
 The first thing that needs to be done is to provision a fairly large CentOS virtual machine top of the Cloud operating system OpenStack.
-This can of course be done via the OpenStack UI or can be automated using our Ansible openstack playbook like so:
 
+This can of course be done via the OpenStack UI or can be automated using our Ansible openstack playbook. For the sake of the examples, and since the VM name will be used around several steps, the `VM_NAME` env variable will store the vm name.
+
+```bash
+$ VM_NAME=vm20210221-t01
 ```
-ansible-playbook playbook/openstack.yml -e type=cloude -e '{"state": "present", "hostname": "somehostname", "openstack": {"os_username": "username", "os_password": "password", "os_domain": "domain", "os_auth_url": "https://somehost:13000/v3", "os_project_id": "someprojectid", "os_network": "network", "os_security_group": "security_group"}}'`
+
+```bash
+$ ansible-playbook playbook/openstack_vm_create_aggregate.yml -e k8s_type=masters -e k8s_version=121 -e '{"openstack": {"vm": {"network": "provider_net_shared"}}}' -e vm_name=${VM_NAME} --tags create
```

Comment:
Why do you use this bizarre wording as I mentioned on the Hetzner PR `openstack_vm_create_aggregate.yml` and not something like `openstack_vm_k8s.yml` or `openstack_vm_k8s_provisioning.yml` ?

Reply:
The idea of these aggregate playbooks is of having a higher level of abstraction reusing existing playbooks. For instance, in order to create a openstack VM we have to execute several playbooks so the aggregate playbook calls all of them.  In this case it calls the following playbooks:
* passstore_controller_check.yml
* passstore_controller_inventory.yml
* openstack_vm_create.yml
* sec_host.yml
for now they don't deploy k8s.

I can call them differently but I wanted to separate them from the "_unitary_" playbooks

Reply:
Keep then the name. The most important is to be sure that when someone will execute the command they will not have to write supeeeeeer long command name. We should perhaps create within the ansible-k8s tooling  image some aliases to help the user to create a vm+k8s or delete a vm+k8s

Reply:
I understand, I'll try finding a better name

Reply:
I'd like that we manage the k8s on another PR as we also have to adapt the installation for new k8s versions.